### ISSUE TITLE
BackBufferCopy regression in 3.2.4-rc1

### ISSUE THREAD

**Issue by lonevox** [godotengine/godot#45838]
**Godot version:**

3.2.4-rc1

**OS/device including version:**

Windows 10
GLES3
NVidia GT 1030 2GB

**Issue description:**

![bug](https://user-images.githubusercontent.com/38600896/107292348-8a3d1c00-6ace-11eb-97d2-27ce296b7df0.png)
I expected the icon to look the same in 3.2.3 and 3.2.4-rc1. However, in 3.2.4-rc1 the BackBufferCopy doesn't seem to work, as the PaletteSwap shader uses the original SCREEN_TEXTURE.

**Steps to reproduce:**
Have any node with a screen space shader, then below it have a BackBufferCopy node, then below that have any other screen space shader. In 3.2.3 they will combine as expected, but in 3.2.4-rc1 the second shader will replace the first.

**Minimal reproduction project:**

[BackBufferCopy_bug.zip](https://github.com/godotengine/godot/files/5947389/BackBufferCopy_bug.zip)

**Comment by Calinou** [comment]
This may be a regression from the unified GLES3/GLES2 batching.

cc @lawnjelly

**Comment by lawnjelly** [comment]
It is batching regression, it looks pixellated when you switch off batching.

I had a look at the diagnose log:
```
canvas_begin FRAME 553
items
	joined_item 1 refs
			batch R 0-1 [0 - 153] {255 255 255 255 }
	joined_item 1 refs
			batch R 0-1 [0 - 0] {255 255 255 255 }
	joined_item 1 refs
			batch R 0-1 [0 - 0] {255 255 255 255 }
	joined_item 1 refs
canvas_end
	items reordered: 1
```
It looks caused by the item reordering, if you set `rendering->batching->parameters->item_reordering_lookahead` to 0, it no longer occurs. I'll see if I can figure out a way to prevent it happening.

**Comment by akien-mga** [comment]
Fixed by #45843.
